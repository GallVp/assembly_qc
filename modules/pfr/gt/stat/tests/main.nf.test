nextflow_process {

    name "Test Process GT_STAT"
    script "../main.nf"
    process "GT_STAT"

    tag "modules"
    tag "modules_nfcore"
    tag "gt"
    tag "gt/stat"

    test("sarscov2-gff3") {

        when {
            process {
                """
                input[0] = Channel.of(
                    "##gff-version 3" +
                    file(params.test_data['sarscov2']['genome']['genome_gff3'], checkIfExists: true).getText().toLowerCase()
                )
                .collectFile(name: 'sample.gff3', newLine: true)
                .map { file -> [ [ id:'test' ], file ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert path(process.out.stats.get(0).get(1)).getText().contains("cdss: 12") }
            )
        }

    }

}