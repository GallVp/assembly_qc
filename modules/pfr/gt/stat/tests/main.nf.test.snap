{
    "sarscov2-gff3": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gt.stat.yml:md5,ebba7831ddbf916b8bbea675ba8693b5"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a184b50afb2ad6dd2d3d37b0a211dd71"
                ],
                "stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gt.stat.yml:md5,ebba7831ddbf916b8bbea675ba8693b5"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a184b50afb2ad6dd2d3d37b0a211dd71"
                ]
            }
        ],
        "timestamp": "2023-11-29T11:34:48.057277"
    }
}